1. Importando pandas e carregando dataset.

In [3]:
import pandas as pd
df = pd.read_csv("netflix1.csv")

2. Exibindo as 10 primeiras linhas do dataframe.

In [4]:
df.head(10)

show_id     type                             title             director  \
0      s1    Movie              Dick Johnson Is Dead      Kirsten Johnson   
1      s3  TV Show                         Ganglands      Julien Leclercq   
2      s6  TV Show                     Midnight Mass        Mike Flanagan   
3     s14    Movie  Confessions of an Invisible Girl        Bruno Garotti   
4      s8    Movie                           Sankofa         Haile Gerima   
5      s9  TV Show     The Great British Baking Show      Andy Devonshire   
6     s10    Movie                      The Starling       Theodore Melfi   
7    s939    Movie   Motu Patlu in the Game of Zones          Suhas Kadav   
8     s13    Movie                      Je Suis Karl  Christian Schwochow   
9    s940    Movie          Motu Patlu in Wonderland          Suhas Kadav   

          country date_added  release_year rating   duration  \
0   United States  9/25/2021          2020  PG-13     90 min   
1          France  9/24/2021          2021  TV-MA   1 Season   
2   United States  9/24/2021          2021  TV-MA   1 Season   
3          Brazil  9/22/2021          2021  TV-PG     91 min   
4   United States  9/24/2021          1993  TV-MA    125 min   
5  United Kingdom  9/24/2021          2021  TV-14  9 Seasons   
6   United States  9/24/2021          2021  PG-13    104 min   
7           India   5/1/2021          2019  TV-Y7     87 min   
8         Germany  9/23/2021          2021  TV-MA    127 min   
9           India   5/1/2021          2013  TV-Y7     76 min   

                                           listed_in  
0                                      Documentaries  
1  Crime TV Shows, International TV Shows, TV Act...  
2                 TV Dramas, TV Horror, TV Mysteries  
3                 Children & Family Movies, Comedies  
4   Dramas, Independent Movies, International Movies  
5                       British TV Shows, Reality TV  
6                                   Comedies, Dramas  
7  Children & Family Movies, Comedies, Music & Mu...  
8                       Dramas, International Movies  
9         Children & Family Movies, Music & Musicals

3. Verificando o tipo de cada coluna e se há colunas ausentes.

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8790 entries, 0 to 8789
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8790 non-null   object
 1   type          8790 non-null   object
 2   title         8790 non-null   object
 3   director      8790 non-null   object
 4   country       8790 non-null   object
 5   date_added    8790 non-null   object
 6   release_year  8790 non-null   int64 
 7   rating        8790 non-null   object
 8   duration      8790 non-null   object
 9   listed_in     8790 non-null   object
dtypes: int64(1), object(9)
memory usage: 686.8+ KB


R: Não há colunas ausentes.

4. Estatísticas descritivas das colunas numéricas.

In [6]:
df.describe()

release_year
count   8790.000000
mean    2014.183163
std        8.825466
min     1925.000000
25%     2013.000000
50%     2017.000000
75%     2019.000000
max     2021.000000

In [7]:
df_duration = df['duration']
df_duration.describe()

count         8790
unique         220
top       1 Season
freq          1791
Name: duration, dtype: object

R: Observa-se a partir da linha "mean" que a maioria dos filmes presentes no dataset foram lançados em 2014. Além disso, o dataset apresenta informações relacionados a filmes de 1925 (min) até 2021 (max). Ademais, a partir da coluna "duration", é possível notar a maior prevalência de conteúdos com apenas uma temporada, sendo eles 1791 séries e filmes com essa duração.

5. Convertendo a coluna 'date_added' para datetime.


In [8]:
df['date_added'] = pd.to_datetime(df['date_added'])

##verificando se foi convertida
df['date_added'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 8790 entries, 0 to 8789
Series name: date_added
Non-Null Count  Dtype         
--------------  -----         
8790 non-null   datetime64[ns]
dtypes: datetime64[ns](1)
memory usage: 68.8 KB


6. Crie as colunas 'month_added' e 'year_added' a partir de 'date_added':


In [9]:
df['month_added'] = df['date_added'].dt.month
df['year_added'] = df['date_added'].dt.year

## verificando se foram criadas
df.describe()

date_added  release_year  month_added   year_added
count                           8790   8790.000000  8790.000000  8790.000000
mean   2019-05-17 21:44:01.638225408   2014.183163     6.655859  2018.873606
min              2008-01-01 00:00:00   1925.000000     1.000000  2008.000000
25%              2018-04-06 00:00:00   2013.000000     4.000000  2018.000000
50%              2019-07-03 00:00:00   2017.000000     7.000000  2019.000000
75%              2020-08-19 18:00:00   2019.000000    10.000000  2020.000000
max              2021-09-25 00:00:00   2021.000000    12.000000  2021.000000
std                              NaN      8.825466     3.436103     1.573568

7. Conferindo quantos títulos há por tipo.


In [10]:
df_movies = df[(df['type'] == 'Movie')]
df_movies.info()

<class 'pandas.core.frame.DataFrame'>
Index: 6126 entries, 0 to 8783
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   show_id       6126 non-null   object        
 1   type          6126 non-null   object        
 2   title         6126 non-null   object        
 3   director      6126 non-null   object        
 4   country       6126 non-null   object        
 5   date_added    6126 non-null   datetime64[ns]
 6   release_year  6126 non-null   int64         
 7   rating        6126 non-null   object        
 8   duration      6126 non-null   object        
 9   listed_in     6126 non-null   object        
 10  month_added   6126 non-null   int32         
 11  year_added    6126 non-null   int32         
dtypes: datetime64[ns](1), int32(2), int64(1), object(8)
memory usage: 574.3+ KB


In [11]:
df_tvshow = df[(df['type'] == 'TV Show')]
df_tvshow.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2664 entries, 1 to 8789
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   show_id       2664 non-null   object        
 1   type          2664 non-null   object        
 2   title         2664 non-null   object        
 3   director      2664 non-null   object        
 4   country       2664 non-null   object        
 5   date_added    2664 non-null   datetime64[ns]
 6   release_year  2664 non-null   int64         
 7   rating        2664 non-null   object        
 8   duration      2664 non-null   object        
 9   listed_in     2664 non-null   object        
 10  month_added   2664 non-null   int32         
 11  year_added    2664 non-null   int32         
dtypes: datetime64[ns](1), int32(2), int64(1), object(8)
memory usage: 249.8+ KB


In [12]:
## ou pode ser feito da seguinte forma:
df['type'].value_counts()

## OBS: Descobri essa solução pesquisando depois.

type
Movie      6126
TV Show    2664
Name: count, dtype: int64

R: Há 6126 filmes e 2664 TV Shows

8. Filtrando somente os filmes e liste aqueles com duração maior que 100 minutos e que ao mesmo tempo foram lançados a partir de 2020:


In [13]:
df['duration_num'] = df['duration'].str.extract(r'(\d+)').astype(int)

df_films100_2020 = df[(df['type'] == 'Movie') & (df['duration_num'] > 100) & (df['release_year'] >= 2020)]
df_films100_2020

show_id   type                               title  \
6        s10  Movie                        The Starling   
8        s13  Movie                        Je Suis Karl   
12      s852  Movie                    99 Songs (Tamil)   
54       s31  Movie                     Ankahi Kahaniya   
55       s36  Movie      The Father Who Moves Mountains   
...      ...    ...                                 ...   
6675    s484  Movie                         Last Summer   
6967   s1595  Movie                   Leyla Everlasting   
7079   s2098  Movie                      Rising Phoenix   
7202   s2356  Movie                          Disclosure   
7311   s2780  Movie  Crip Camp: A Disability Revolution   

                                               director         country  \
6                                        Theodore Melfi   United States   
8                                   Christian Schwochow         Germany   
12                                            Not Given        Pakistan   
54    Ashwiny Iyer Tiwari, Abhishek Chaubey, Saket C...       Not Given   
55                                         Daniel Sandu       Not Given   
...                                                 ...             ...   
6675                                          Not Given        Pakistan   
6967                                          Not Given          Turkey   
7079                                          Not Given  United Kingdom   
7202                                          Not Given   United States   
7311                                          Not Given   United States   

     date_added  release_year rating duration  \
6    2021-09-24          2021  PG-13  104 min   
8    2021-09-23          2021  TV-MA  127 min   
12   2021-05-21          2021  TV-14  131 min   
54   2021-09-17          2021  TV-14  111 min   
55   2021-09-17          2021  TV-MA  110 min   
...         ...           ...    ...      ...   
6675 2021-07-09          2021  TV-MA  102 min   
6967 2020-12-04          2020  TV-MA  113 min   
7079 2020-08-26          2020  PG-13  107 min   
7202 2020-06-19          2020  TV-MA  108 min   
7311 2020-03-25          2020      R  108 min   

                                              listed_in  month_added  \
6                                      Comedies, Dramas            9   
8                          Dramas, International Movies            9   
12       Dramas, International Movies, Music & Musicals            5   
54     Dramas, Independent Movies, International Movies            9   
55              Dramas, International Movies, Thrillers            9   
...                                                 ...          ...   
6675      Dramas, International Movies, Romantic Movies            7   
6967                     Comedies, International Movies           12   
7079  Documentaries, International Movies, Sports Mo...            8   
7202                        Documentaries, LGBTQ Movies            6   
7311                                      Documentaries            3   

      year_added  duration_num  
6           2021           104  
8           2021           127  
12          2021           131  
54          2021           111  
55          2021           110  
...          ...           ...  
6675        2021           102  
6967        2020           113  
7079        2020           107  
7202        2020           108  
7311        2020           108  

[339 rows x 13 columns]

9. Separando a coluna 'duration' em 'duration_int' (numérico) e 'duration_unit'
(unidade):

In [14]:
df[['duration_int', 'duration_unit']] = df['duration'].str.extract(r'(\d+)\s*(\w+)')
df['duration_int'] = df['duration_int'].astype(int)

## verificando se a separação ocorreu
df

show_id     type                             title         director  \
0         s1    Movie              Dick Johnson Is Dead  Kirsten Johnson   
1         s3  TV Show                         Ganglands  Julien Leclercq   
2         s6  TV Show                     Midnight Mass    Mike Flanagan   
3        s14    Movie  Confessions of an Invisible Girl    Bruno Garotti   
4         s8    Movie                           Sankofa     Haile Gerima   
...      ...      ...                               ...              ...   
8785   s8797  TV Show                        Yunus Emre        Not Given   
8786   s8798  TV Show                         Zak Storm        Not Given   
8787   s8801  TV Show                Zindagi Gulzar Hai        Not Given   
8788   s8784  TV Show                              Yoko        Not Given   
8789   s8786  TV Show                               YOM        Not Given   

            country date_added  release_year rating   duration  \
0     United States 2021-09-25          2020  PG-13     90 min   
1            France 2021-09-24          2021  TV-MA   1 Season   
2     United States 2021-09-24          2021  TV-MA   1 Season   
3            Brazil 2021-09-22          2021  TV-PG     91 min   
4     United States 2021-09-24          1993  TV-MA    125 min   
...             ...        ...           ...    ...        ...   
8785         Turkey 2017-01-17          2016  TV-PG  2 Seasons   
8786  United States 2018-09-13          2016  TV-Y7  3 Seasons   
8787       Pakistan 2016-12-15          2012  TV-PG   1 Season   
8788       Pakistan 2018-06-23          2016   TV-Y   1 Season   
8789       Pakistan 2018-06-07          2016  TV-Y7   1 Season   

                                              listed_in  month_added  \
0                                         Documentaries            9   
1     Crime TV Shows, International TV Shows, TV Act...            9   
2                    TV Dramas, TV Horror, TV Mysteries            9   
3                    Children & Family Movies, Comedies            9   
4      Dramas, Independent Movies, International Movies            9   
...                                                 ...          ...   
8785                  International TV Shows, TV Dramas            1   
8786                                           Kids' TV            9   
8787  International TV Shows, Romantic TV Shows, TV ...           12   
8788                                           Kids' TV            6   
8789                                           Kids' TV            6   

      year_added  duration_num  duration_int duration_unit  
0           2021            90            90           min  
1           2021             1             1        Season  
2           2021             1             1        Season  
3           2021            91            91           min  
4           2021           125           125           min  
...          ...           ...           ...           ...  
8785        2017             2             2       Seasons  
8786        2018             3             3       Seasons  
8787        2016             1             1        Season  
8788        2018             1             1        Season  
8789        2018             1             1        Season  

[8790 rows x 15 columns]

10. Criando a coluna ‘seasons_int’ e colocando o valor numérico da quantidade de
temporada contida em “duration” para as linhas TV Show.


In [15]:
df['seasons_int'] = None
df.loc[df['type'] == 'TV Show', 'seasons_int'] = df.loc[df['type'] == 'TV Show', 'duration_num']
df['seasons_int']

0       None
1          1
2          1
3       None
4       None
        ... 
8785       2
8786       3
8787       1
8788       1
8789       1
Name: seasons_int, Length: 8790, dtype: object

11. Mostrando quais são a série com mais temporadas e o filme com maior tempo
de duração.


In [16]:
more_seasons = df[df['type'] == 'TV Show'].sort_values(by='seasons_int', ascending=False).head(1)

more_duration = df[df['type'] == 'Movie'].sort_values(by='duration_int', ascending=False).head(1)
more_seasons

show_id     type           title   director        country date_added  \
6697    s549  TV Show  Grey's Anatomy  Not Given  United States 2021-07-03   

      release_year rating    duration                     listed_in  \
6697          2020  TV-14  17 Seasons  Romantic TV Shows, TV Dramas   

      month_added  year_added  duration_num  duration_int duration_unit  \
6697            7        2021            17            17       Seasons   

     seasons_int  
6697          17

In [17]:
more_duration

show_id   type                       title   director        country  \
7822   s4254  Movie  Black Mirror: Bandersnatch  Not Given  United States   

     date_added  release_year rating duration  \
7822 2018-12-28          2018  TV-MA  312 min   

                                           listed_in  month_added  year_added  \
7822  Dramas, International Movies, Sci-Fi & Fantasy           12        2018   

      duration_num  duration_int duration_unit seasons_int  
7822           312           312           min        None

R: Grey´s Anatomy e Black Mirror: Bandersnatch.

12. Agrupando por 'rating' e contando quantos títulos existem em cada categoria.


In [18]:
df.groupby('type')['title'].count()

type
Movie      6126
TV Show    2664
Name: title, dtype: int64

R: 6126 filmes e 2664 séries/tv shows.

13. Identificando os 5 diretores com maior número de títulos:


In [19]:
titles_per_director = df.groupby('director').size().sort_values(ascending=False).head(5)
titles_per_director

director
Not Given                 2588
Rajiv Chilaka               20
Raúl Campos, Jan Suter      18
Alastair Fothergill         18
Suhas Kadav                 16
dtype: int64

R: Not Given (provalmente Netflix), Rajiv Chilaka, Raúl Campos e Jan Sueter empatados com o mesmo número de títulos, Alastair Fothergill e Suhas Kadav indicam os diretores com maior número de títulos.

14.Formule uma hipótese subjetiva sobre o dataset.

In [29]:
movies_df = df[df['type'] == 'Movie'].copy()
movies_df['short_movie'] = movies_df['duration_int'] < 90
short_movies_per_year = movies_df.groupby('year_added')['short_movie'].sum()

short_movies_per_year

year_added
2008      1
2009      1
2010      1
2011      1
2012      3
2013      4
2014     12
2015     37
2016    129
2017    318
2018    371
2019    410
2020    327
2021    222
Name: short_movie, dtype: int64

R: Mediante análise do dataset, é possível analisar a crescente no número de filmes mais curtos, contendo duração de menos de 1h30min, ao longo dos anos, tangendo dos anos 2008 a 2021. Tal ocorrência pode ser justificada pelo aumento no uso da tecnologia e redes sociais, que podem refletir em uma possível tendência por contéudos mais rápidos de se consumir.